app/src/main/java/com/iic/shopingo/api/BaseApiRequest.java

```diff
@@ -0,0 +1,51 @@
+package com.iic.shopingo.api;
+
+import android.util.Log;
+import bolts.Task;
+import java.util.concurrent.Callable;
+
+/**
+ * Created by asafg on 11/03/15.
+ */
+public abstract class BaseApiRequest<T extends ApiResult> {
+
+  protected String authToken;
+  protected Server server;
+
+  public BaseApiRequest(String authToken) {
+    this.authToken = authToken;
+    server = new Server(authToken);
+  }
+
+  public abstract T executeSync();
```

Comment:
By default, all methods are assumed to be synchronous, you don't have `toStringSync`.